packages/client/src/workflow-client.ts

```diff
@@ -0,0 +1,580 @@
+      }
+      throw new errors.WorkflowExecutionContinuedAsNewError('Workflow execution continued as new', newExecutionRunId);
+    }
+  }
+
+  /**
+   * Uses given input to make a queryWorkflow call to the service
+   *
+   * Used as the final function of the query interceptor chain
+   */
+  protected async _queryWorkflowHandler(input: WorkflowQueryInput): Promise<unknown> {
+    const response = await this.service.queryWorkflow({
+      queryRejectCondition: input.queryRejectCondition,
+      namespace: this.options.namespace,
+      execution: input.workflowExecution,
+      query: {
+        queryType: input.queryType,
+        queryArgs: { payloads: this.options.dataConverter.toPayloads(...input.args) },
+      },
+    });
+    if (response.queryRejected) {
+      if (response.queryRejected.status === undefined || response.queryRejected.status === null) {
+        throw new TypeError('Received queryRejected from server with no status');
+      }
+      throw new QueryRejectedError(response.queryRejected.status);
+    }
+    if (!response.queryResult) {
+      throw new TypeError('Invalid response from server');
+    }
+    // We ignore anything but the first result
+    return this.options.dataConverter.fromPayloads(0, response.queryResult?.payloads);
+  }
+
+  /**
+   * Uses given input to make a signalWorkflowExecution call to the service
+   *
+   * Used as the final function of the signal interceptor chain
+   */
+  protected async _signalWorkflowHandler(input: WorkflowSignalInput): Promise<void> {
+    console.log('signalWorkflowExecution', input.workflowExecution);
+    const resp = await this.service.signalWorkflowExecution({
+      identity: this.options.identity,
+      namespace: this.options.namespace,
+      workflowExecution: input.workflowExecution,
+      requestId: uuid4(),
+      // control is unused,
+      signalName: input.signalName,
+      input: { payloads: this.options.dataConverter.toPayloads(...input.args) },
+    });
+    console.log({ resp });
+  }
+
+  /**
+   * Uses given input to make terminateWorkflowExecution call to the service
+   *
+   * Used as the final function of the terminate interceptor chain
+   */
+  protected async _terminateWorkflowHandler(
+    input: WorkflowTerminateInput
+  ): Promise<TerminateWorkflowExecutionResponse> {
+    return this.service.terminateWorkflowExecution({
+      namespace: this.options.namespace,
+      identity: this.options.identity,
+      ...input,
+      details: { payloads: this.options.dataConverter.toPayloads(input.details) },
+    });
+  }
+
+  /**
+   * Create a {@link WorkflowStub} for a new Workflow execution
+   *
+   * @param name workflow type name (the filename in the Node.js SDK)
+   * @param options used to start the Workflow
+   */
+  public stub<T extends Workflow>(name: string, options: WorkflowOptions): WorkflowStub<T>;
+
+  /**
+   * Create a {@link WorkflowStub} for an existing Workflow execution
```

Comment:
Technically slightly misleading. "Workflow execution" usually means a specific run. I would just say existing workflow here. To be more specific you can say it will target the most recent run of the provided WF id

Reply:
👍